processor/spanmetricsprocessor/processor.go

```diff
@@ -257,20 +289,20 @@ func (p *processorImp) collectLatencyMetrics(ilm pdata.InstrumentationLibraryMet
 		dpLatency.SetStartTimestamp(pdata.NewTimestampFromTime(p.startTime))
 		dpLatency.SetTimestamp(timestamp)
 		dpLatency.SetExplicitBounds(p.latencyBounds)
-		dpLatency.SetBucketCounts(p.latencyBucketCounts[key])
-		dpLatency.SetCount(p.latencyCount[key])
-		dpLatency.SetSum(p.latencySum[key])
+		dpLatency.SetBucketCounts(p.latencyBucketCounts[resAttrKey][mKey])
+		dpLatency.SetCount(p.latencyCount[resAttrKey][mKey])
+		dpLatency.SetSum(p.latencySum[resAttrKey][mKey])
 
-		setLatencyExemplars(p.latencyExemplarsData[key], timestamp, dpLatency.Exemplars())
+		setLatencyExemplars(p.latencyExemplarsData[resAttrKey][mKey], timestamp, dpLatency.Exemplars())
 
-		p.metricKeyToDimensions[key].CopyTo(dpLatency.Attributes())
+		p.metricKeyToDimensions[mKey].CopyTo(dpLatency.Attributes())
 	}
 }
 
 // collectCallMetrics collects the raw call count metrics, writing the data
 // into the given instrumentation library metrics.
-func (p *processorImp) collectCallMetrics(ilm pdata.InstrumentationLibraryMetrics) {
-	for key := range p.callSum {
+func (p *processorImp) collectCallMetrics(ilm pdata.InstrumentationLibraryMetrics, resAttrKey resourceKey) {
+	for mKey := range p.callSum[resAttrKey] {
```

Comment:
I think this is the same as well, you'll need to hold a lock here.

Reply:
Yeah this is fine